src/types/common.d.ts

```diff
@@ -3,7 +3,7 @@ import { CommonHintsArgs } from '../hint';
 import { AbstractDataType, DataType } from '../data_types';
 import { AbstractBone } from './abstract_bone';
 
-export type Literal = null | undefined | boolean | number | bigint | string | Date | object | ArrayBuffer;
+export type Literal = null | undefined | boolean | number | bigint | string | Date | Record<string, any> | ArrayBuffer;
```

Comment:
bigint types gets checked correctly since `object` type is removed from Literal union type